pyramid/pyramid.go

```diff
@@ -0,0 +1,22 @@
+package pyramid
+
+// StorageType is enumeration of the different tiered storage types
+// supported by pyramid.
+type StorageType int
+
+const (
+	StorageTypeSSTable StorageType = iota
+	StorageTypeTreeManifest
+)
+
+// FS is pyramid abstraction of filesystem where the persistent storage-layer is the block storage.
+// Files on the Local disk are transient and might be cleaned up by the eviction policy.
+type FS interface {
+	// Store uploads the file to the block storage.
+	// It may remove the file from the local storage.
+	Store(t StorageType, filename string) error
```

Comment:
Can you give a good way to get a useful filename to use here?  I _do_ want my client to use "the right" directory.  E.g. if I pick my own filename it may be on the wrong filesystem.

Reply:
Nope, you can't reference files from a different filesystem, that's guaranteed.